Site Settings: Remove post type toggle when theme supports it

tyxla opened the pull request:
This PR updates the Custom Content Types card to hide the toggles for a certain post type if that post type is already supported by the active theme. This is necessary, because even though the toggle works fine in that case (saves and retrieves the value properly), it doesn't reflect whether the corresponding post type is displayed in the sidebar. This PR is part of #13568.

Before:
![](https://cldup.com/pCi6OHbiFO.png)

After:
![](https://cldup.com/m2Ib_T9gSe.png)

Note: if a post type is not supported for a theme, the toggle will still be shown for that post type, as it is on the "Before" screenshot. The "After" screenshot is from a site that uses a theme that supports both the Testimonials and the Portfolio post types.

To test:
* Checkout this branch or get it going on calypso.live.
* Go to `/settings/writing/$site`, where `$site` is one of your Jetpack sites (wp.com sites will be supported too when we land #14572).
* Use a theme that supports both post the Testimonial and Portfolio post types.
* Verify the toggles are not displayed.
* Use a theme that supports one of the Testimonial and Portfolio post types.
* Verify only the toggle of the non-supported post type is displayed.
* Verify the post type toggles work as they did before.

oskosk commented:
LGTM

oskosk commented:
@tyxla, I labeled this as ready to merge, but feel free to remove the label if Design proposes changes.

sirreal commented:
Inverted control enable/disable:

![testimonial-enabled](https://cloud.githubusercontent.com/assets/841763/26586004/29f8118c-454e-11e7-8b0f-8977d9f0be11.png)

tyxla commented:
@sirreal this is expected behavior. See instructions above:

>> Verify only the toggle of the non-supported post type is displayed.

sirreal commented:
@tyxla I misread 😬 

LGTM!

rickybanister commented:
As it stands I understand @sirreal's confusion. I was going to suggest we add some kind of compact notice here that says 'your theme already supports these', but realized that is unnecessary. 

I think once we add the 'post per page' setting to this block (#13868) it will make these settings less 'read-only' so this is fine for now.

tyxla commented:
Thanks for the review, y'all! 🙇 

> I think once we add the 'post per page' setting to this block (#13868) it will make these settings less 'read-only'

Yes, that's exactly the plan, I just wanted to approach this task granularly with small PRs.

> so this is fine for now.

Thanks, I'll go ahead and merge then.

MichaelArestad commented:
Could we leave the toggles and instead switch them to on and disabled or is that somehow misleading?

rickybanister commented:
I like your thinking, but that's likely the wrong solution. We could use the green checkmark that we're using for the always-on stats module and others—that's kind of our indicator of 'this is working'.

tyxla commented:
@MichaelArestad @rickybanister the green checkmark sounds like a good solution, I've updated the PR with it, here is an updated preview:

![](https://cldup.com/Ff2sGtQPeO.png)

rickybanister commented:
The checkmark is great BUT the description text might create confusion with the `if your theme does not support them use the shortcode` text. Can we conditionalize that part of the description without bodging up the translations and whatnot? It'd probably be good to swap that sentence for `Your theme has built-in testimonial support.`

rickybanister commented:
@sirreal I believe you're right. We weren't taking advantage of the detection method when that copy was written originally. 

Let's go with your label suggestions, and just make sure we link the words 'testimonial' and 'portfolio' to the support documents.

tyxla commented:
Thanks @sirreal @rickybanister! I've updated the wording as per the suggestions:

![](https://cldup.com/pFtSepxYYy.png)

![](https://cldup.com/VZVopiebt8.png)

Let me know if that looks good @rickybanister. Also @sirreal that one could use another code review, as I've changed some small bits.

sirreal commented:
Toggles are changing to checks. Preferably order matches sidebar:

![lose-toggle2](https://cloud.githubusercontent.com/assets/841763/26726744/7fb67e26-47a4-11e7-950a-adbd65003de4.gif)

Here they both flip on at the same time:

![lose-toggle](https://cloud.githubusercontent.com/assets/841763/26726743/7fb1662a-47a4-11e7-92dd-5c9b247487b2.gif)

tyxla commented:
Seems like we're not able to properly differentiate whether a post type support comes from the theme or not. To be able to, we'll need something similar to what we need in #13637. I'm marking this one as "In Progress" for the time being as we're working on it.

tyxla commented:
This one'll be done in a little different way, so I'm closing this PR.
